Error when used in overnightjs project

Caused by incorrect placing of parentheses during the `isArray` check.

**Existing**
https://github.com/express-promise-router/express-promise-router/blob/6bf785ae1f04e12cb209745642d377cb99a80020/lib/express-promise-router.js#L80

**Modified**
```js
(Array.isArray(args[0]) && ('string' === typeof args[0][0] || args[0][0] instanceof RegExp))
```

Possibly relates to https://github.com/express-promise-router/express-promise-router/issues/46

That appeared to solve that problem, however a new one developed, please
take a look at the error message

    throw new Error(msg);
        ^

Error: Expected a callback function but got a [object Undefined]
    at wrapHandler
(C:\Users\Acer\Documents\React\chatW\server\ApiAuthentication\node_modules\express-promise-router\lib\express-promise-router.js:10:15)
    at
C:\Users\Acer\Documents\React\chatW\server\ApiAuthentication\node_modules\express-promise-router\lib\express-promise-router.js:87:24
    at Array.map (<anonymous>)
    at Route.instanceToWrap.(anonymous function) [as get]
(C:\Users\Acer\Documents\React\chatW\server\ApiAuthentication\node_modules\express-promise-router\lib\express-promise-router.js:86:38)
    at Object.<anonymous>
(C:\Users\Acer\Documents\React\chatW\server\ApiAuthentication\routes\users.js:14:6)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! apiauthentication@1.0.0 server: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the apiauthentication@1.0.0 server script.
npm ERR! This is probably not a problem with npm. There is likely
additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!
C:\Users\Acer\AppData\Roaming\npm-cache\_logs\2020-03-19T14_52_26_076Z-debug.log

On Thu, Mar 19, 2020 at 3:50 PM Solomon Unoke <soulstar205@gmail.com> wrote:

> I see it now, thanks, I'll apply the change and hopefully everything works
> out
>
>
> On Thu, Mar 19, 2020 at 3:25 PM Andrew Mackrodt <notifications@github.com>
> wrote:
>
>> Your error trace shows you haven't made the modification:
>>
>> Yours:
>> (Array.isArray(args[0]) && 'string' === typeof args[0][0] || args[0][0]
>> instanceof RegExp)
>>
>> Expected:
>> (Array.isArray(args[0]) && ('string' === typeof args[0][0] || args[0][0]
>> instanceof RegExp))
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/express-promise-router/express-promise-router/issues/64#issuecomment-601208359>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AF6HPMQRLJ3AZM2FCVBV5PDRIITNRANCNFSM4IKK6C3A>
>> .
>>
>

--- a/.changeset/late-crews-admire.md
+++ b/.changeset/late-crews-admire.md
@@ -1,0 +1,5 @@
+---
+"express-promise-router": patch
+---
+
+Fix incorrect array access check.

--- a/lib/express-promise-router.js
+++ b/lib/express-promise-router.js
@@ -79,8 +79,8 @@ var wrapMethods = function (instanceToWrap, isRoute) {
       if (
         "string" === typeof args[0] ||
         args[0] instanceof RegExp ||
-        (Array.isArray(args[0]) && "string" === typeof args[0][0]) ||
-        args[0][0] instanceof RegExp
+        (Array.isArray(args[0]) &&
+          ("string" === typeof args[0][0] || args[0][0] instanceof RegExp))
       ) {
         first = args[0];
         args = args.slice(1);

